File: app/src/main/java/br/pedroso/upcomingmovies/movieslist/di/MoviesComponent.java
package br.pedroso.upcomingmovies.movieslist.di;

import br.pedroso.upcomingmovies.di.ApplicationComponent;
import br.pedroso.upcomingmovies.di.scopes.FragmentScope;
import br.pedroso.upcomingmovies.movieslist.ui.MoviesActivity;
import dagger.Component;

@FragmentScope
@Component(dependencies = ApplicationComponent.class, modules = MoviesPresenterModule.class)
public interface MoviesComponent {
    void inject(MoviesActivity activity);
}

File: app/src/main/kotlin/br/pedroso/upcomingmovies/movieslist/di/MoviesComponent.kt
package br.pedroso.upcomingmovies.movieslist.di

import br.pedroso.upcomingmovies.di.ApplicationComponent
import br.pedroso.upcomingmovies.di.scopes.FragmentScope
import br.pedroso.upcomingmovies.movieslist.MoviesListActivity
import dagger.Component

@FragmentScope
@Component(dependencies = [ApplicationComponent::class], modules = [MoviesPresenterModule::class])
interface MoviesComponent {
    fun inject(activity: MoviesListActivity)
}
